### PR TITLE
Validation of reordered readonly parameters in setParameters

### PR DESCRIPTION
**Pull request by aboba**
Fix for Issue https://github.com/w3c/webrtc-pc/issues/1492

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -5597,8 +5597,9 @@ <h2>Methods</h2>
                 <code>InvalidStateError</code>.</li>
                 <li>If <code><var>parameters</var>.encodings.length</code>
                 is different from <var>N</var>, or if
-                any parameter in <var>parameters</var>, is
-                marked as a <dfn>Read-only parameter</dfn>, has a value that is
+                <code><var>parameters</var>.encodings</code> has been re-ordered,
+                or if any parameter in <var>parameters</var> is
+                marked as a <dfn>Read-only parameter</dfn> or has a value that is
```
Comment (taylor-b) [inline webrtc.html]: I think "or has a value" is meant to be "and has a value"? In other words, "is read-only *and* was changed"?

Comment (stefhak) [reply webrtc.html]: Yes, agree.

Comment (alvestrand) [reply webrtc.html]: Might be better to rephrase - the sequence "A or B and C" isn't necessarily parsed as "A or (B and C)".

File: webrtc.html [hunk 2]
```diff
@@ -5595,17 +5595,21 @@ <h2>Methods</h2>
...
+                <code>InvalidModificationError</code>:
+                <ol>  
+                   <li><code><var>parameters</var>.encodings.length</code>
+                   is different from <var>N</var>.</li>
+                   <li>Any parameter in <var>parameters</var> is
+                   marked as a <dfn>Read-only parameter</dfn> and has a value that is
+                   different from the corresponding parameter value in <var>sender</var>'s
+                   <a>[[\LastReturnedParameters]]</a> internal slot. Note that this also
+                   applies to <var>transactionId</var>.</li>
+                   <li><code><var>parameters</var>.encodings</code> has been re-ordered.</li>
```
Comment (taylor-b) [inline webrtc.html]: nit: May want to swap the second and third conditions, so that the ones dealing with `parameters.encodings` are next to each other.

**Pull request merged**